### PR TITLE
Feature: file folder pretty view

### PR DESCRIPTION
**Pull request by MaksGovor**

### CODE REVIEW

File: src/shared/util.ts [hunk 1]
```diff
@@ -6,3 +6,10 @@ export const hashCognitoSecret = (clientSecret: string, username: string, client
 		.update(username + clientId)
 		.digest('base64');
 };
+
+export const getBasename = (str: string, sep = '/'): string => {
+	const index = str.lastIndexOf(sep);
+	const basename = str.slice(index + 1);
+	if (!basename && index + 1 === str.length) return getBasename(str.slice(0, index), sep);
+	return basename;
+};
```
Comment (YAGoOaR) [inline src/shared/util.ts:15]:
```suggestion
```
Use "path" lib instead

File: src/pages/FilesPage/components/FolderItem/FolderItem.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import Typography from '@mui/material/Typography';
+import {
+	FolderItemCard,
+	FolderItemCheckbox,
+	FolderItemContainer,
+	FolderItemIcon,
+} from './FolderItem.styles';
+import { FolderResponse } from 'clients/CoreService';
+import { renderLongName } from '../renderers/renderLongName';
+import { getBasename } from 'shared/util';
```
Comment (YAGoOaR) [inline src/pages/FilesPage/components/FolderItem/FolderItem.tsx:11]:
```suggestion
import { basename } from 'path';
```
Use "path" lib instead

File: src/pages/FilesPage/components/FolderItem/FolderItem.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		if (checked) {
+			setChecked(ids => {
+				return ids.filter(basename => basename !== folder.path);
+			});
+		} else {
+			setChecked(checked => [...checked, folder.path]);
+		}
+	};
+
+	const folderName: string = getBasename(folder.path);
```
Comment (YAGoOaR) [inline src/pages/FilesPage/components/FolderItem/FolderItem.tsx:39]:
```suggestion
	const folderName: string = basename(folder.path);
```
Use "path" lib instead

File: src/pages/FilesPage/FilesPage.tsx [hunk 4]
```diff
@@ -6,158 +6,82 @@ import Divider from '@mui/material/Divider';
...
 function FilesPage() {
-	const coreUrl: string = process.env.REACT_APP_CORE_URL || '';
-	const queryClient = useQueryClient();
+	const [currentLocation, setCurrentLocation] = useState<string>('');
 	const [processingFiles, setProcessingFiles] = useState<Array<FileInfo>>([]);
-	const [checkedIds, setCheckedIds] = useState<Array<number>>([]);
+	const [checkedFolders, setCheckedFolders] = useState<Array<string>>([]);
+	const [checkedFiles, setCheckedFiles] = useState<Array<string>>([]);
 
 	const { data: existingFiles, isLoading: isFilesLoading } = useQuery(
```
Comment (tedi4t) [inline src/pages/FilesPage/FilesPage.tsx:27]: Please, rename it. It contains not only files, but it also contains folders.

File: src/pages/FilesPage/FilesPage.tsx [hunk 5]
```diff
@@ -6,158 +6,82 @@ import Divider from '@mui/material/Divider';
...
 import { FileInfo } from './components/FileItem/FileItem';
+import { FolderItem } from './components/FolderItem';
+import { ControlMenu } from './components/ControlMenu';
+import { LocationLinks } from './components/LocationLinks';
 
 function FilesPage() {
-	const coreUrl: string = process.env.REACT_APP_CORE_URL || '';
-	const queryClient = useQueryClient();
+	const [currentLocation, setCurrentLocation] = useState<string>('');
 	const [processingFiles, setProcessingFiles] = useState<Array<FileInfo>>([]);
```
Comment (tedi4t) [inline src/pages/FilesPage/FilesPage.tsx:23]: We don't use setProcessingFiles anywhere, are you sure that is will be fine?

File: src/pages/FilesPage/components/FolderItem/FolderItem.styles.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import styled from 'styled-components';
+import FolderIcon from '@mui/icons-material/Folder';
+import Card from '@mui/material/Card';
+import Checkbox from '@mui/material/Checkbox';
+import Box from '@mui/material/Box';
+
+export const FolderItemIcon = styled(FolderIcon)`
+	font-size: 100px;
+	color: #6e52b3;
```
Comment (tedi4t) [inline src/pages/FilesPage/components/FolderItem/FolderItem.styles.ts:9]: We use this color at least two times for now. We can set colors for our @mui-ui theme and we will just write "primary", "secondary" and so on. We also have to do the same thing for font-sizes.

File: src/pages/FilesPage/components/FolderItem/FolderItem.styles.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import styled from 'styled-components';
+import FolderIcon from '@mui/icons-material/Folder';
+import Card from '@mui/material/Card';
+import Checkbox from '@mui/material/Checkbox';
+import Box from '@mui/material/Box';
+
+export const FolderItemIcon = styled(FolderIcon)`
+	font-size: 100px;
```
Comment (tedi4t) [inline src/pages/FilesPage/components/FolderItem/FolderItem.styles.ts:8]: Please, use rem instread of px

File: src/pages/FilesPage/components/FileItem/FileItem.styles.ts [hunk 8]
```diff
@@ -1,13 +1,20 @@
...
 
-export const WCard = styled(Card)`
+export const FileIcon = styled(InsertDriveFileIcon)`
+	font-size: 90px;
+	color: #6e52b3;
+	margin-bottom: 6px;
+`;
+
+export const FileCard = styled(Card)`
+	display: inline-block;
```
Comment (tedi4t) [inline src/pages/FilesPage/components/FileItem/FileItem.styles.ts:14]: Why don't you use flex instead? What will happen if we have several rows?

File: src/pages/FilesPage/components/FolderItem/FolderItem.tsx (hunk 3 above)
Comment (tedi4t) [inline src/pages/FilesPage/components/FolderItem/FolderItem.tsx:39]: We return folder name, don't we?

File: src/pages/FilesPage/FilesPage.tsx (hunk 5 above)
Comment (MaksGovor) [reply src/pages/FilesPage/FilesPage.tsx:23]: We do not have requests on the backend for this yet, as soon as they appear I will use it, everything is under control)

Comment (tedi4t) [reply src/pages/FilesPage/FilesPage.tsx:23]: ok)

File: src/pages/FilesPage/components/FolderItem/FolderItem.tsx (hunk 3 above)
Comment (MaksGovor) [reply src/pages/FilesPage/components/FolderItem/FolderItem.tsx:39]: From the backend we need the full path for the user, to display we need only the name. getBasename function on the frontend for this

File: src/pages/FilesPage/components/FileItem/FileItem.styles.ts (hunk 8 above)
Comment (MaksGovor) [reply src/pages/FilesPage/components/FileItem/FileItem.styles.ts:14]: There is a grid container on top and I wrap these components in the grid item, but you're right, this style is not needed here at all

File: src/pages/FilesPage/FilesPage.tsx [hunk 9]
```diff
@@ -1,55 +1,58 @@
...
-
-import {
-	FileMenuButton,
-	FilesContainer,
-	FilesGridContainer,
-	FlexBox,
-	UHeader,
-} from './FilePage.styles';
+import { FilesContainer, FilesGridContainer } from './FilePage.styles';
+import theme from 'themes/theme';
```
Comment (tedi4t) [inline src/pages/FilesPage/FilesPage.tsx:9]:
```suggestion
import theme from 'themes';
```

File: src/pages/FilesPage/FilesPage.tsx [hunk 10]
```diff
@@ -1,55 +1,58 @@
...
 	const queryClient = useQueryClient();
+	const [currentLocation, setCurrentLocation] = useState<string>('');
 	const [processingFiles, setProcessingFiles] = useState<Array<FileInfo>>([]);
-	const [checkedIds, setCheckedIds] = useState<Array<number>>([]);
+	const [checkedItems, setCheckedItems] = useState<Array<string>>([]);
 
-	const { data: existingFiles, isLoading: isFilesLoading } = useQuery(
-		[entities.file],
-		() => FileService.list(),
+	const { data: existingContent, isLoading: isContentLoading } = useQuery(
```
Comment (tedi4t) [inline src/pages/FilesPage/FilesPage.tsx:30]:
```suggestion
	const { data: content, isLoading: isContentLoading } = useQuery(
```

File: src/pages/FilesPage/FilesPage.tsx [hunk 11]
```diff
@@ -1,55 +1,58 @@
...
 
-					await queryClient.invalidateQueries(entities.file);
-					setProcessingFiles(processingFiles =>
-						processingFiles?.filter(processedFile => processedFile.name !== file.name),
-					);
+			return Promise.all(
+				urls.map((postUrl: PostUrlInfo) => {
+					const fileName = getBasename(postUrl.path);
+					const file = files.find(file => file.name === fileName);
+					if (!file) throw new Error('Bad link generation');
```
Comment (tedi4t) [inline src/pages/FilesPage/FilesPage.tsx:50]: create variable `onError = useSnackbarOnError()` upper and do `onError(new Error('Bad link generation'))` here

File: src/pages/FilesPage/FilesPage.tsx [hunk 12]
```diff
@@ -69,26 +77,28 @@ function FilesPage() {
...
+				names: filePathes.map(filePath => getBasename(filePath)),
+			};
+			return FileManagerService.getSignedGetUrls(request);
+		},
 		{
 			onError: useSnackbarOnError(),
-			onSuccess: (response: FileLinkResponse) => {
-				window.open(coreUrl + response.link);
-			},
+			onSuccess: downloadFilesByLink,
```
Comment (tedi4t) [inline src/pages/FilesPage/FilesPage.tsx:101]:
```suggestion
			onSuccess: downloadFilesFromSignedUrls,
```

**Review by tedi4t — APPROVED** [review]

**Pull request merged**